packages/engine/lib/provider/src/lib.rs

```diff
@@ -110,6 +110,8 @@
 //
 //   TODO: replace library with https://github.com/rust-lang/project-error-handling/issues/3.
 
+#![warn(clippy::pedantic, clippy::nursery)]
```

Comment:
+1 for these as warnings. I don't think we should ever block on them, but it's good to get the visibility (provided folks are able to make sensible decisions about what to address/when, and can avoid the distraction of their likely omnipresence).

Reply:
Actually we block on any warning currently which is strictly not backwards compatible: https://github.com/hashintel/hash/blob/1cfc2e7965905dd3e934b993c13252d235475799/.github/workflows/rust.yml#L154

I have written this crate taking into account those warnings. Adding those to our current engine would be just noise. We even disabled some of the most annoying clippy lints, but as we proceed, we want to enable them again:
https://github.com/hashintel/hash/blob/1cfc2e7965905dd3e934b993c13252d235475799/packages/engine/src/lib.rs#L22-L35